user_tools/src/spark_rapids_tools/tools/qualx/util.py

```diff
@@ -221,6 +227,43 @@ def compute_accuracy(
     return scores
 
 
+def compute_precision_recall(results: pd.DataFrame, y: str, y_preds: Dict[str, str], threshold: float):
+    """Compute precision and recall from a dataframe using a threshold for identifying true positives.
+
+    Parameters
+    ----------
+    results: pd.DataFrame
+        Pandas dataframe containing the label column and one or more prediction columns.
+    y: str
+        Label column name.
+    y_preds: Dict[str, str]
+        Dictionary of display name of prediction -> prediction column name.
+    threshold: float
+        Threshold separating positives (inclusive of threshold) from negatives.
+
+    Returns
+    -------
+    precision: Dict[str, float]
+        Dictionary of precision metric per prediction column,
+        e.g. {'QX': 0.90, 'QXS': 0.92}
+    recall: Dict[str, float]
+        Dictionary of recall metric per prediction column,
+        e.g. {'QX': 0.78, 'QXS': 0.82}
+    """
+    precision = {}
+    recall = {}
+    for name, y_pred in y_preds.items():
+        tp = sum((results[y_pred] >= threshold) & (results[y] >= threshold))
+        fp = sum((results[y_pred] >= threshold) & (results[y] < threshold))
+        # tn = sum((results[y_pred] < threshold) & (results[y] < threshold))
```

Comment:
nit: Is the unused line kept for future usage? Do we want to plot the confusion matrix?

Reply:
Was mostly just for completeness, so removed for now.